strider/caching.py

```diff
@@ -34,18 +36,69 @@ async def cache_wrapper(*args, **kwargs):
     return cache_wrapper
 
 
-def async_locking_cache(fcn):
-    """Cache decorator."""
+def async_locking_query_cache(fcn, maxsize=32):
```

Comment:
I think that we can make this a generic function fairly easily, and that we should prefer to keep this generic. I think it could be confusing as is because this function doesn't actually care if it's a query it's caching. To make this generic we can replace `hash_query(args[0])` with `json.dumps( (args,kwargs) )`

Reply:
Updated in f80179b